interface/src/app/sample.service.js

```diff
@@ -74,8 +76,43 @@ angular.module('adage.sample.service', [
           $log.error($scope.analysis.queryStatus);
         }
       ).$promise;
+      Sample.promises[pk] = pSample;
       return pSample;
     };
+    Sample.getSampleListPromise = function(pkList) {
```

Comment:
Can you add a blank line between lines 81 and 82? I think it will make the code more readable.